docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/BuildingsProcessor.php

```diff
@@ -65,6 +111,32 @@ protected function processRecord(&$record) {
     }
   }
 
+  protected function processResult(&$result) {
+    try {
+      $building_image_url = $result->imageUrl;
+      $this->client->request('GET', $building_image_url);
+
+      $scheme = $this->configFactory->get('system.file')->get('default_scheme');
+      $destination = $scheme . '://building_images/';
+      $building_number = $result->buildingNumber;
+      $realpath = $this->fs->realpath($destination);
+
+      if ($this->fs->prepareDirectory($realpath, FileSystemInterface::CREATE_DIRECTORY)) {
+        /** @var FileInterface $file */
+        $file = system_retrieve_file($building_image_url, "{$destination}{$building_number}.jpg", TRUE, FileSystemInterface::EXISTS_REPLACE);
+        $result->imageUrl = $file->getFileUri();
```

Comment:
```suggestion
        $data = file_get_contents($building_image_url);
        $file = \Drupal::service('file.repository')->writeData($data, "{$destination}{$building_number}.jpg", FileSystemInterface::EXISTS_REPLACE);
        $result->imageUrl = ['target_id' => $file->id()];
```
I think this is a way to rewrite this to generate a file which can be referenced with a file field.